API/models/schemas/refs/types.json

```diff
@@ -2,34 +2,48 @@
     "$id": "refs/types.json",
     "definitions": {
         "color": {
-            "pattern": "^[0-9a-fA-F]{6}$"
+            "pattern": "^[0-9a-fA-F]{6}$",
+            "description": "a 6 length hex value (e.g. 'C0FFEE')"
+        },
+        "colorTemplate": {
+            "pattern": "(^[0-9a-fA-F]{6}$)|(^@[a-zA-Z0-9-]+$)|(^$)",
+            "description": "a 6 length hex value or '@[color name]' (e.g. 'C0FFEE', '@blue')"
         },
         "metricImperialUnit": {
-            "enum": ["mm", "cm", "m", "f"]
+            "enum": ["mm", "cm", "m", "f"],
+            "description": "'mm', 'cm', 'm' or 'f'"
         },
         "metricImperialUUnit": {
-            "enum": ["mm", "cm", "m", "f", "U", "OU"]
+            "enum": ["mm", "cm", "m", "f", "U", "OU"],
+            "description": "'mm', 'cm', 'm', 'f', 'U' or 'OU'"
         },
         "vector2": {
-            "pattern": "^\\s*{\\s*\"x\"\\s*:\\s*([-|+]?[0-9]*[.]?)?[0-9]+\\s*,\\s*\"y\"\\s*:\\s*([-|+]?[0-9]*[.]?)?[0-9]+\\s*}\\s*$"
+            "pattern": "^\\s*{\\s*\"x\"\\s*:\\s*((-|\\+)?[0-9]*[.]?)?[0-9]+\\s*,\\s*\"y\"\\s*:\\s*((-|\\+)?[0-9]*[.]?)?[0-9]+\\s*}\\s*$",
+            "description": "a vector2 (e.g. '{\"x\": 2.5, \"y\": 3}', '{\"x\": -3.5, \"y\": -3}')"
         },
         "vector3": {
-            "pattern": "^\\s*{\\s*\"x\"\\s*:\\s*([-|+]?[0-9]*[.]?)?[0-9]+\\s*,\\s*\"y\"\\s*:\\s*([-|+]?[0-9]*[.]?)?[0-9]+\\s*,\\s*\"z\"\\s*:\\s*([-|+]?[0-9]*[.]?)?[0-9]+\\s*}\\s*$"
+            "pattern": "^\\s*{\\s*\"x\"\\s*:\\s*((-|\\+)?[0-9]*[.]?)?[0-9]+\\s*,\\s*\"y\"\\s*:\\s*((-|\\+)?[0-9]*[.]?)?[0-9]+\\s*,\\s*\"z\"\\s*:\\s*((-|\\+)?[0-9]*[.]?)?[0-9]+\\s*}\\s*$",
+            "description": "a vector3 (e.g. '{\"x\": 2, \"y\": 2.5, \"z\": 3.33}', '{\"x\": 2, \"y\": -4, \"z\": -3.5}')"
         },
         "float": {
-            "pattern": "^[-]?([0-9]*[.])?[0-9]+$"
+            "pattern": "^[-]?([0-9]*[.])?[0-9]+$",
+            "description": "a floating point number (e.g. '3.14', '-2.72', '1', '-2')"
         },
         "clearanceVector": {
-            "pattern": "^\\[(?:\\d+(?:\\.\\d*)?|\\.\\d+)(?:,\\s*(?:\\d+(?:\\.\\d*)?|\\.\\d+)){5}\\]$"
+            "pattern": "^\\[(?:\\d+(?:\\.\\d*)?|\\.\\d+)(?:,\\s*(?:\\d+(?:\\.\\d*)?|\\.\\d+)){5}\\]$",
+            "description": "six floating point numbers inside square brackets (e.g. '[2, 2.72, 3, 3.14, 4, 4.5]'"
         },
         "slug": {
-            "pattern": "^[a-z0-9-_]+$"
+            "pattern": "^[a-z0-9-_]+$",
+            "description": "a pattern with lowercase letters, numbers, '-' and '_' (e.g. 'abc_xyz-0')"
```

Comment:
This gives me the impression that it's mandatory to have all of them: lowercase, number, - and _

Reply:
I've changed the messages to make it clearer :)